crates/ide-completion/src/item.rs

```diff
@@ -269,6 +292,27 @@ impl CompletionRelevance {
         if is_definite {
             score += 10;
         }
+
+        score += associated_fn
+            .map(|asf| {
+                let mut score = match asf.ty {
+                    CompletionRelevanceFnType::ReturnsExpectedType => 20,
+                    CompletionRelevanceFnType::DirectConstructor => 15,
+                    CompletionRelevanceFnType::Builder => 10,
+                    CompletionRelevanceFnType::Constructor => 5,
+                    CompletionRelevanceFnType::Other => 0,
+                };
+
+                // Prefer functions with no arguments, then functions with self arguments
+                if score > 0 {
+                    score += if !asf.has_args { 2 } else { 0 };
+                    score -= if asf.has_self_arg { score - 1 } else { 0 };
```

Comment:
```suggestion
                    score -= if asf.has_self_arg { 1 } else { 0 };
```

This sets score to `-1`, looks like a mistake. Though subtracting is risky since we start with `score = 0` which could underflow

Reply:
There was a check ` if score > 0 {` to ensure it never underflows, however I've rewrite this for clarity this time.